### PR TITLE
make artifacts contain magic.h and libmagic.lib

### PR DESCRIPTION
**Pull request by Cirn09**
I want to write a windows program that depends on libmagic(`magic.h` and `libmagic.lib`), so I added `install` to CMakeLists.txt.

### CODE REVIEW

**Comment by Cirn09** [comment]
`libmagic` can be used as a static library, but `DllMain` is defined in https://github.com/file/file/blob/87731415de/src/magic.c#L163.
`DllMain` only initialized `_w32_dll_instance`. 
`_w32_dll_instance` is only used in [`get_default_magic`](https://github.com/file/file/blob/87731415de/src/magic.c#L239).
 In fact, when `libmagic` is used as a static library, only https://github.com/file/file/blob/87731415de/src/magic.c#L236 is enough.

File: .gitignore [hunk 1]
```diff
@@ -84,7 +84,7 @@ x64/
 CMakeFiles/
 *.vcxproj
 *.filters
-*.cmake
+; *.cmake
```
Comment (julian-r) [inline .gitignore:87]: Why committing the `.cmake` files?

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,82 +7,109 @@ function(cat IN_FILE OUT_FILE)
...
+option(BUILD_SHARED_LIBS "build as shared library" ON)
+
+# set(FILE_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/file/src")
+set(LIBMAGIC_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/file/src")
+set(LIBMAGIC_TEST_DIR "${CMAKE_CURRENT_SOURCE_DIR}/file/tests")
+set(WIN_HEADERS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/win-headers")
+# set(PCRE2_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/pcre2/src")
+set(PCRE2_LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/pcre2/install_dir/lib")
+
+set(FILE_VERSION 5.38)
```
Comment (julian-r) [inline CMakeLists.txt:23]: That would be a downgrade of file to `5.38`, is there a specific reason for this?

**Review by julian-r — CHANGES_REQUESTED** [review]

**Comment by Cirn09** [comment]
Sorry, I noticed your comment on my code only after you closed this PR. I will rewrite the code and open a new PR.

For developers who use this project as a dynamic or static library, the `install` command at the end of the CMakeLists.txt file is necessary. Otherwise, cmake will not generate `.cmake` files for this project as a library. The `cmake/Findpcre2.cmake` file is also necessary, otherwise the result of `cmake install` will not include the content of pcre2.

Also, the suggestion of compiling file as a static library under Windows has been merged into the file source code. Therefore, the implementation solution in this project can be simplified significantly.

**Pull request closed without merging**